lib/core/asynctree.js

```diff
@@ -78,9 +78,8 @@ class AsyncTree extends EventEmitter{
 
         if (result instanceof Error) {
           abortOnFailure = result.abortOnFailure || Utils.isUndefined(result.abortOnFailure);
-        } else {
-          node.resolve(result);
         }
+        node.resolve(result);
```

Comment:
this is actually fixed in v2 by rejecting the promise is case of an error: https://github.com/nightwatchjs/nightwatch/blob/v2/lib/core/asynctree.js#L80

Reply:
`node.reject` causes the script to end prematurely unless the test script is handling the unhandled rejection.

Reply:
ah yes, we removed it in #2802.

Reply:
But does the unhadled rejection error also occur when using assert? It sounds to me that if we always resolve the promise here the test execution will also continue when there's a failure, regardless of the value of abortOnFailure. 

Also, there are some test failures in this PR.

Reply:
I checked that case. The script ends at the assert reporting the error. I will check the tests that are failing